scenarios/sns_secrets/terraform/api_gateway.tf

```diff
@@ -1,92 +1,87 @@
-  rest_api_id = aws_api_gateway_rest_api.cg_api.id
-  parent_id   = aws_api_gateway_rest_api.cg_api.root_resource_id
-  path_part   = "resource-${var.cgid}"
+resource "aws_api_gateway_resource" "resource" {
+  rest_api_id = aws_api_gateway_rest_api.api.id
+  parent_id   = aws_api_gateway_rest_api.api.root_resource_id
+  path_part   = "user-data"
 }
 
-resource "aws_api_gateway_method" "cg_method" {
-  rest_api_id   = aws_api_gateway_rest_api.cg_api.id
-  resource_id   = aws_api_gateway_resource.cg_resource.id
-  http_method   = "GET"
-  authorization = "NONE"
+resource "aws_api_gateway_method" "method" {
+  rest_api_id      = aws_api_gateway_rest_api.api.id
+  resource_id      = aws_api_gateway_resource.resource.id
+  http_method      = "GET"
+  authorization    = "NONE"
   api_key_required = true
 }
 
-resource "aws_api_gateway_integration" "cg_integration" {
-  rest_api_id = aws_api_gateway_rest_api.cg_api.id
-  resource_id = aws_api_gateway_resource.cg_resource.id
-  http_method = aws_api_gateway_method.cg_method.http_method
+resource "aws_api_gateway_integration" "integration" {
+  rest_api_id = aws_api_gateway_rest_api.api.id
+  resource_id = aws_api_gateway_resource.resource.id
+  http_method = aws_api_gateway_method.method.http_method
   type        = "MOCK"
+
   request_templates = {
-    "application/json" = "{\"statusCode\": 200}"
+    "application/json" = jsonencode({
+      statusCode = 200
+    })
   }
 }
 
-resource "aws_api_gateway_method_response" "cg_method_response" {
-  rest_api_id = aws_api_gateway_rest_api.cg_api.id
-  resource_id = aws_api_gateway_resource.cg_resource.id
-  http_method = aws_api_gateway_method.cg_method.http_method
+resource "aws_api_gateway_method_response" "method_response" {
+  rest_api_id = aws_api_gateway_rest_api.api.id
+  resource_id = aws_api_gateway_resource.resource.id
+  http_method = aws_api_gateway_method.method.http_method
   status_code = "200"
 }
 
-resource "aws_api_gateway_integration_response" "cg_integration_response" {
-  rest_api_id = aws_api_gateway_rest_api.cg_api.id
-  resource_id = aws_api_gateway_resource.cg_resource.id
-  http_method = aws_api_gateway_method.cg_method.http_method
-  status_code = aws_api_gateway_method_response.cg_method_response.status_code
+resource "aws_api_gateway_integration_response" "integration_response" {
+  rest_api_id = aws_api_gateway_rest_api.api.id
+  resource_id = aws_api_gateway_resource.resource.id
+  http_method = aws_api_gateway_method.method.http_method
+  status_code = aws_api_gateway_method_response.method_response.status_code
   response_templates = {
-    "application/json" = <<EOF
-    {
-      "message": "Access granted.",
-      "user_data": {
-        "user_id": "1337",
-        "username": "SuperAdmin",
-        "email": "SuperAdmin@notarealemail.com",
-        "password": "p@ssw0rd123"
-      },
-      "final_flag": "FLAG{SNS_S3cr3ts_ar3_FUN}"
-    }
-    EOF
+    "application/json" = jsonencode({
```

Comment:
Switched this to be json encoded
1. Looks prettier ;)
2. More of an expect output from an API since it will remove newlines